### PR TITLE
US441

### PR DESCRIPTION
**Pull request by dzadroga**
Completes US441 which is a centralized strategy for handling purchases
Tasks:
- Create StandardPurchaseStrategy with GetPurchaseDecision(tile, player, state)
- Integrate PurchaseStrategy with TileResolver and Ownership/EconomyManager
- Support multiple purchase flows such as no offer, offer to player, action hook even if placeholder logic
- Write tests to make sure purchase logic and ownership updates are working properly
- Create a class diagram for property, ownership, and purchase strategy interactions

All tests created pass

### CODE REVIEW

File: Assets/Scripts/Managers/Purchase/StandardPurchaseStrategy.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                return new PurchaseDecision { Flow = PurchaseFlow.None };
+
+            //Checks if already owned by someone else
+            //If so then No purchase
+            var currentOwner = own.GetOwner(tile);
+            if (currentOwner != null && currentOwner != buyer)
+                return new PurchaseDecision { Flow = PurchaseFlow.None };
+
+            //Placeholder pricing: 4x base rent 
+            //We can add the real purchase price later
```
Comment (hdathert) [inline Assets/Scripts/Managers/Purchase/StandardPurchaseStrategy.cs]: In the `OwnableSpaceData` types, we already have the buy price as a field in the asset, we should be able to get that from there, via whatever adapter system we're using (`ITileRentInfo` I guess)

File: Assets/Tests/EditMode/PurchaseManagerTests/StandardPurchaseStrategyTests.cs [hunk 2]
```diff
@@ -0,0 +1,131 @@
+using NUnit.Framework;
+using UnityEngine;
+using Assets.Scripts.Managers.Rent;
+using Assets.Scripts.Managers.Purchase;
+
+namespace Tests.EditMode.PurchaseTests
+{
+    public class StandardPurchaseStrategyTests
+    {
+        private class Tile : ITileRentInfo
```
Comment (hdathert) [inline Assets/Tests/EditMode/PurchaseManagerTests/StandardPurchaseStrategyTests.cs]: Since we're using effectively this struct in multiple files, it seems like maybe that should live in its own utility class.

File: UMLDiagrams/Task449.mmd [hunk 3]
```diff
@@ -0,0 +1,119 @@
+classDiagram
+
+    class Space {
+        <<abstract>>
+        +name: string
+        +index: int
+        +land(p: Player): void
+    }
+
+    class OwnableSpace {
```
Comment (hdathert) [inline UMLDiagrams/Task449.mmd]: Is this referencing the `OwnableSpaceData` class or some other data class?

**Review by hdathert — CHANGES_REQUESTED** [review]
Just have a few questions on this one.

File: UMLDiagrams/Task449.mmd (hunk 3 above)
Comment (dzadroga) [reply UMLDiagrams/Task449.mmd]: I updated the diagram to use OwnableSpaceData label

File: Assets/Tests/EditMode/PurchaseManagerTests/StandardPurchaseStrategyTests.cs (hunk 2 above)
Comment (dzadroga) [reply Assets/Tests/EditMode/PurchaseManagerTests/StandardPurchaseStrategyTests.cs]: Moved this to its own utility class, all tests are currently passing

File: Assets/Scripts/Managers/Purchase/StandardPurchaseStrategy.cs (hunk 1 above)
Comment (dzadroga) [reply Assets/Scripts/Managers/Purchase/StandardPurchaseStrategy.cs]: The program is now getting data from the adapter systems, there are some sections that I left with TODO comments until we get further into the special rules

**Review by hdathert — APPROVED** [review]
# Code Review Checklist

**Completed for**: US #441 and Tasks #445 to #449  
**Code Review Completed By**: @hdathert 

Use this checklist to review new code changes before merging.

---

## Implementation
- [x] Does this code change accomplish what it is supposed to do?
- [x] Can the solution be simplified?
- [x] This change does **not** add unwanted compile-time or run-time dependencies.
- [x] The code is at the right abstraction level.
- [x] The code is modular enough.
- [x] The solution is adequate for maintainability, readability, performance, and security.
- [x] Similar functionality does **not** already exist in the codebase (or reuse is not appropriate).
- [x] No additional best practices or design patterns would substantially improve this code.

---

## Logic Errors and Bugs
- [x] No use cases where the code behaves incorrectly.
- [x] No inputs or external events that can break the code.
- [x] Code agrees with specifications.
- [x] No off-by-one errors.
- [x] No magic numbers (or they are well-named constants).
- [x] Proper use of global/large-scope variables (minimized and justified).

---

## Error Handling and Logging
- [x] Error handling is implemented correctly.
- [x] Error messages are user-friendly (where applicable).
- [x] Sufficient, meaningful log events exist for debugging (no sensitive data in logs).

---

## Dependencies
- [x] Documentation/configuration/README updates were made as needed.
- [x] No negative impact on other parts of the system or backward compatibility.

---

## Security and Data Privacy
- [x] No new security vulnerabilities introduced.
- [x] All user input is validated, sanitized, and escaped (XSS/SQL injection prevented).
- [x] Data from external APIs/libraries is handled safely and checked as needed.

---

## Performance
- [x] No significant performance regressions.
- [x] No obvious opportunities for significant performance gains left unaddressed.

---

## Ethics and Morality
- [x] Appropriate measures exist to prevent or report harassment/abuse (if applicable).

---

## Testing and Testability
- [x] Code is testable.
- [x] Automated tests have been added or updated.
- [x] Existing tests sufficiently cover the change (unit/integration/system).
- [x] Edge cases and boundary inputs are tested.

---

## Readability
- [x] Code is easy to understand.
- [x] Methods/functions are appropriately small and focused.
- [x] Names (functions/classes/variables) are clear and descriptive.
- [x] Code is in the correct file/folder/package.
- [x] Control flow is intuitive; data flow is clear.
- [x] Comments are clear, relevant, and up to date.
- [x] No commented-out or redundant code remains.
- [x] No overly dense one-liners; no overly long methods.

---

**Use this before submitting a pull request.**  
It helps ensure code quality, maintainability, and compliance with best practices.

*Based on awesome code review guides and MIT’s code review recommendations.*

**Review by tsaldrid — APPROVED** [review]
# Code Review Checklist

**Completed for**: US #441 and Tasks #445 TO #449
**Code Review Completed By**: @tsaldrid

Use this checklist to review new code changes before merging.

---

## Implementation
- [x] Does this code change accomplish what it is supposed to do?
- [x] Can the solution be simplified?
- [x] This change does **not** add unwanted compile-time or run-time dependencies.
- [x] The code is at the right abstraction level.
- [x] The code is modular enough.
- [x] The solution is adequate for maintainability, readability, performance, and security.
- [x] Similar functionality does **not** already exist in the codebase (or reuse is not appropriate).
- [x] No additional best practices or design patterns would substantially improve this code.

---

## Logic Errors and Bugs
- [x] No use cases where the code behaves incorrectly.
- [x] No inputs or external events that can break the code.
- [x] Code agrees with specifications.
- [x] No off-by-one errors.
- [x] No magic numbers (or they are well-named constants).
- [x] Proper use of global/large-scope variables (minimized and justified).

---

## Error Handling and Logging
- [x] Error handling is implemented correctly.
- [x] Error messages are user-friendly (where applicable).
- [x] Sufficient, meaningful log events exist for debugging (no sensitive data in logs).

---

## Dependencies
- [x] Documentation/configuration/README updates were made as needed.
- [x] No negative impact on other parts of the system or backward compatibility.

---

## Security and Data Privacy
- [x] No new security vulnerabilities introduced.
- [x] All user input is validated, sanitized, and escaped (XSS/SQL injection prevented).
- [x] Data from external APIs/libraries is handled safely and checked as needed.

---

## Performance
- [x] No significant performance regressions.
- [x] No obvious opportunities for significant performance gains left unaddressed.

---

## Ethics and Morality
- [x] Appropriate measures exist to prevent or report harassment/abuse (if applicable).

---

## Testing and Testability
- [x] Code is testable.
- [x] Automated tests have been added or updated.
- [x] Existing tests sufficiently cover the change (unit/integration/system).
- [x] Edge cases and boundary inputs are tested.

---

## Readability
- [x] Code is easy to understand.
- [x] Methods/functions are appropriately small and focused.
- [x] Names (functions/classes/variables) are clear and descriptive.
- [x] Code is in the correct file/folder/package.
- [x] Control flow is intuitive; data flow is clear.
- [x] Comments are clear, relevant, and up to date.
- [x] No commented-out or redundant code remains.
- [x] No overly dense one-liners; no overly long methods.

---

**Use this before submitting a pull request.**  
It helps ensure code quality, maintainability, and compliance with best practices.

*Based on awesome code review guides and MIT’s code review recommendations.*
